sunpy/util/tests/test_config.py

```diff
@@ -80,3 +86,32 @@ def test_get_and_create_sample_dir():
     # Set the config back
     os.rmdir(new_path)
     config.set('downloads', 'sample_dir', os.path.join(USER, 'sunpy', 'data', 'sample_data'))
+
+
+def test_write_config(tmpdir, undo_config_dir_patch):
+    config_filename = 'sunpyrc'
+
+    tmp_config_dir = tmpdir.mkdir("config_here")
+    os.environ["SUNPY_CONFIGDIR"] = tmp_config_dir.strpath
+
+    assert tmp_config_dir == _get_user_configdir()
+
+    module_dir = Path(sunpy.__file__).parent
+    config_file = module_dir / 'data' / config_filename
+    user_config_dir = tmp_config_dir
+    user_config_file = user_config_dir / config_filename
+
+    tmp_file = tmpdir.join("hello.txt")
+
+    # Have to write to the file otherwise its seen as a directory(?!)
+    tmp_file.write("content")
+
+    write_config()
+    assert open(user_config_file, 'r').read() == open(config_file, 'r').read()
+
+    with pytest.warns(SunpyUserWarning):
+        write_config(overwrite=False)
+
+    if not (os.name == "nt"):
```

Comment:
This if statement is a bit mysterious. Could you add a clarifying comment?

Reply:
I've added the comments that were there in a similar test above.
Let me know if this needs more clarification. I'll change the comments in both places.